### ISSUE TITLE
org.eclipse.hono:hono-legal:jar:2.6.0-SNAPSHOT (absent)

### ISSUE THREAD

**Issue by ramonunibz** [eclipse-hono/hono#3615]
Hello everyone,

I am a Master's student at the University of Innsbruck, and for my Master's Thesis, I am developing a simulated (for now) digital twin. However, I am encountering an issue while trying to run an example found under `examples/hono-client-examples/src/main/java/org/eclipse/hono/vertx/example`. 

When I execute the command `mvn clean package`, I get the following error:

[ERROR] Failed to execute goal on project hono-example-client: Could not resolve dependencies for project org.eclipse.hono:hono-example-client:jar:2.6.0-SNAPSHOT: The following artifacts could not be resolv
ed: org.eclipse.hono:hono-legal:jar:2.6.0-SNAPSHOT (absent), org.eclipse.hono:hono-client-application-amqp:jar:2.6.0-SNAPSHOT (absent), org.eclipse.hono:hono-client-application-kafka:jar:2.6.0-SNAPSHOT (abs
ent), org.eclipse.hono:hono-client-device-amqp:jar:2.6.0-SNAPSHOT (absent), org.eclipse.hono:hono-demo-certs:jar:2.6.0-SNAPSHOT (absent): Could not find artifact org.eclipse.hono:hono-legal:jar:2.6.0-SNAPSHOT -> [Help 1]

Does anyone know what I am missing? Thank you in advance for you help!

**Comment by sophokles73** [comment]
You need to run `mvn clean install` in the root folder first. This will compile all code into jar files and add them to your local Maven repository. After that, compiling the example(s) should work as expected.

**Comment by ramonunibz** [comment]
Thank you a lot for the fast response! After having done that, I switched back to the hono-example-client folder and ran `mvn clean package`, which worked perfectly. However, when now running the command `mvn exec:java -Dexec.mainClass=org.eclipse.hono.vertx.example.HonoExampleApplication`, I get the following error message:

[ERROR] Unknown lifecycle phase ".mainClass=org.eclipse.hono.vertx.example.HonoExampleApplication". You must specify a valid lifecycle phase or a goal in the format <plugin-prefix>:<goal> or <plugin-group-i
d>:<plugin-artifact-id>[:<plugin-version>]:<goal>. Available lifecycle phases are: pre-clean, clean, post-clean, validate, initialize, generate-sources, process-sources, generate-resources, process-resource
s, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy, pre-site, site, post-site, site-deploy. -> [Help 1]

I have then tried running the command:
`mvn exec:java -Dexec:mainClass=org.eclipse.hono.vertx.example.HonoExampleApplication`, where the difference is that I replaced the "." with ":" after `-Dexec`. Unfortunately, in this case, I ran into the following error:

[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.2.0:java (default-cli) on project hono-example-client: The parameters mainClass for goal org.codehaus.mojo:exec-maven-plugin:3.2.0:java are missing or invalid -> [Help 1]

Thank you again for your help!
